Api/Data/CreditmemoInterface.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2022 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Api\Data;
+
+interface CreditmemoInterface
+{
+    /**
+     * Constants for keys of data array. Identical to the name of the getter in snake case.
+     */
+    const ENTITY_ID = 'entity_id';
+    const PSPREFERENCE = 'pspreference';
+    const ORIGINAL_REFERENCE = 'original_reference';
+
+    const CREDITMEMO_ID = 'creditmemo_id';
+    const ADYEN_ORDER_PAYMENT_ID = 'adyen_order_payment_id';
+    const AMOUNT = 'amount';
+    const STATUS = 'status';
+    const CREATED_AT = 'created_at';
+    const UPDATED_AT = 'updated_at';
+
+    /**
+     * Gets the ID for the creditmemo.
+     *
+     * @return int|null Entity ID.
+     */
+    public function getEntityId(): ?int;
+
+    /**
+     * Sets entity ID.
+     *
+     * @param int $entityId
+     * @return $this
+     */
+    public function setEntityId($entityId): CreditmemoInterface;
+
+    /**
+     * Gets the Pspreference for the creditmemo(capture)
+     *
+     * @return int|null Pspreference.
+     */
+    public function getPspreference(): ?int;
```

Comment:
Can we change this to a string since this can be alphanumeric?
```suggestion
    public function getPspreference(): ?string;
```